core/src/wallet/bitcoin/BitcoinLikeAccount.cpp

```diff
@@ -708,5 +707,195 @@ namespace ledger {
+                            if (derivationPath.nonEmpty()) {
+                                paths.push_back(std::make_shared<DerivationPathApi>(DerivationPath(derivationPath.getValue())));
+                                pub_keys.push_back(self->getKeychain()->getPublicKey(outputAddress).getValue());
+                            }
+                        }
+                        auto input = std::shared_ptr<BitcoinLikeWritableInputApi>(
+                                new BitcoinLikeWritableInputApi(
+                                        self->getExplorer(),
+                                        self->getContext(),
+                                        0xfffffff0,
+                                        pub_keys,
+                                        paths,
+                                        outputAddress,
+                                        std::make_shared<Amount>(
+                                                self->getKeychain()->getCurrency(),
+                                                0,
+                                                BigInt(newOutput->getValue()->toBigInt()->toString(10))
+                                        ),
+                                        newOutput->getTransactionHash(),
+                                        newOutput->getOutputIndex(),
+                                        {},
+                                        std::make_shared<BitcoinLikeOutputApi>(
+                                                newOutput->getOutput(),
+                                                self->getWallet()->getCurrency()
+                                        )
+                                )
+                        );
+                        tx->addInput(input);
+                        utxo++;
+                    }
+
+
+                    // Adjust change output
+                    auto additionalChange = aggregatedAmount - additionalFees;
+                    if (outputChange) {
+                        auto changeOutput = std::dynamic_pointer_cast<BitcoinLikeOutputApi>(outputChange);
+                        auto out = changeOutput->getOutput();
+                        out.value = out.value + BigInt((unsigned long long) additionalChange);
+                        tx->addOutput(
+                                std::make_shared<BitcoinLikeOutputApi>(
+                                        out,
+                                        self->getWallet()->getCurrency()
+                                )
+                        );
+                        tx->removeOutput(outputChange->getOutputIndex());
+                    } else {
+                        // Create a change
+                        auto changeAddress = self->getKeychain()->getFreshAddress(BitcoinLikeKeychain::CHANGE)->toString();
+                        BitcoinLikeBlockchainExplorerOutput out;
+                        out.index = static_cast<uint64_t>(tx->getOutputs().size());
```

Comment:
I’m just realizing that all that “create a new change output” is a bit `unsafe` / _doing the same thing over and over_. I’m not asking any change for that PR but I’ll keep the idea in mind for our next iteration of libcore. We should have somekind of middleware / abstraction that does that work for us so that we don’t even have to manipulate the keychain logic.

Reply:
> Very good PR, thank you! Just a few questions:
> 
> * I am right to guess that currently, that PR doesn’t allow a user to confirm a RBF-signaled TX?
> * Currently, we do not check whether a transaction is confirmed when we retrieve them. So it means that we might spend unconfirmed transactions.
> * On the same leve, does it mean that we display unconfirmed amounts to the user on a `getBalance`?

What do you mean by confirming an RBF-signaled tx ? 
If you are talking about additional UTXOs that we might need, no we don't check but we take deeper first, so worst case all UTXOs are generated by unconfirmed txs and we can't do better than that in that case,
Yes, because of the optimistic update, but again if a tx got dropped the mechanism of sync is removing the tx from our db and the `getBalance` will return the new correct balance